src/unix/linux.c

```diff
@@ -2270,6 +2270,136 @@ uint64_t uv_get_available_memory(void) {
 }
 
 
+static int uv__get_cgroupv2_constrained_cpu(const char* cgroup, 
+                                            uv__cpu_constraint* constraint) {
+  char path[256];
+  char buf[1024];
+  unsigned int weight;
+  int cgroup_size;
+  const char* cgroup_trimmed;
+  char quota_buf[16];
+
+  if (strncmp(cgroup, "0::/", 4) != 0)
+    return UV_EINVAL;
+   
+  /* Trim ending \n by replacing it with a 0 */
+  cgroup_trimmed = cgroup + sizeof("0::/") - 1;      /* Skip the prefix "0::/" */
+  cgroup_size = (int)strcspn(cgroup_trimmed, "\n");  /* Find the first slash */
+
+  /* Construct the path to the cpu.max file */
+  snprintf(path, sizeof(path), "/sys/fs/cgroup/%.*s/cpu.max", cgroup_size,
+           cgroup_trimmed);
+
+  /* Read cpu.max */
+  if (uv__slurp(path, buf, sizeof(buf)) < 0)
+    return UV_EIO;
+
+  if (sscanf(buf, "%15s %llu", quota_buf, &constraint->period_length) != 2)
+    return UV_EINVAL;
+
+  if (strcmp(quota_buf, "max") == 0)
+    constraint->quota_per_period = LLONG_MAX;
+  else
+    constraint->quota_per_period = strtoll(quota_buf, NULL, 10);
```

Comment:
Should we check for errors in the conversion?

Reply:
Is this the correct way to handle errors ?

```c
  if (strncmp(quota_buf, "max", 3) == 0)
    constraint->quota_per_period = LLONG_MAX;
  else {
    char* endptr;
    errno = 0;
    constraint->quota_per_period = strtoll(quota_buf, &endptr, 10);
    if ((errno == ERANGE && (constraint->quota_per_period == LLONG_MAX || constraint->quota_per_period == LLONG_MIN)) || (errno != 0 && constraint->quota_per_period == 0)) {
        return UV_EINVAL; // overflow
    }
    if (*endptr != '\0')
      return UV_EINVAL; // conversion failed
  }
```

Reply:
Or maybe simpler

```c
  if (strncmp(quota_buf, "max", 3) == 0)
    constraint->quota_per_period = LLONG_MAX;
  else if (sscanf(quota_buf, "%lld", &constraint->quota_per_period) != 1)
    return UV_EINVAL; // conversion failed
```